Backend/index.js

```diff
@@ -39,14 +40,53 @@ app.use(
   cors({
     origin: "http://localhost:5173",
     credentials: true,
-    methods: ["GET", "POST", "DELETE"],
+    methods: ["GET", "POST", "DELETE", "PUT"],
   })
 );
 
 io.on("connection", (socket) => {
-  socket.on("disconnect", () => {});
+  const userId = socket.handshake.query.userId;
+  // Mark user as online
+  socket.join(userId);
+  socket.on("disconnect", () => {
+    socket.leave(userId);
```

Comment:
what is this new logic doing?

Reply:
When a user connects, the server retrieves their userId from the WebSocket handshake query. This makes them online and adds them to a room with their user id, then disconnects them when they go offline